lib/puppet_x/bodeco/util.rb

```diff
@@ -25,10 +25,15 @@ def initialize(_url, options)
         @password = options[:password]
         @cookie = options[:cookie]
         @insecure = options[:insecure]
-        proxy_server = options[:proxy_server]
-        proxy_type = options[:proxy_type]
 
-        ENV["#{proxy_type}_proxy"] = proxy_server
+        if options[:proxy_server]
+          uri = URI(options[:proxy_server])
+          unless uri.scheme
+            uri = URI("#{options[:proxy_type]}://#{options[:proxy_server]}")
```

Comment:
This line should never be hit?

I still think we've got confusion about what `proxy_type` is supposed to do.  At the moment, it behaves differently depending on the provider.  Then according to the README and original proxy PR, `proxy_server` is supposed to be of the form `http://my_proxy.example.com:3128`  I think it's confusing to have some providers treat the parameters differently.